core/storage/btree.rs

```diff
@@ -4453,7 +4453,11 @@ impl BTreeCursor {
     /// ```
     ///
     /// The destruction order would be: [4',4,5,2,6,7,3,1]
-    pub fn btree_destroy(&mut self) -> Result<CursorResult<()>> {
+    pub fn btree_destroy(&mut self, former_root_page: &mut usize) -> Result<CursorResult<()>> {
```

Comment:
could this function return an `Option<usize>` instead of taking in a `&mut usize`?

Reply:
Yeah that definitely resembles Rust semantics better.

addressed by 4312d371fb9c1c3821781259e9f1b3de590c5bed